plugins/cpp_metrics/parser/src/cppmetricsparser.cpp

```diff
@@ -98,6 +100,52 @@ bool CppMetricsParser::cleanupDatabase()
   return true;
 }
 
+
+void CppMetricsParser::afferentCouplingTypeLevel()
+{
+  util::OdbTransaction{_ctx.db}([&,this]
+  {
+    std::set<std::uint64_t> typesFound;
+    std::unordered_map<std::uint64_t, int> typeFoundCnt;
+    std::unordered_map<std::uint64_t,std::uint64_t> astNodeIdOfType;
+
+    for (const model::AfferentRecordView& type
+        : _ctx.db->query<model::AfferentRecordView>())
+    {
+      if (!cc::util::isRootedUnderAnyOf(_inputPaths, type.filePath))
+      {
+        continue;
+      }
+
+      typesFound.clear();
+      for (const model::CppMemberType& member : _ctx.db->query<model::CppMemberType>(
+        odb::query<cc::model::CppMemberType>::typeHash == type.entityHash &&
+        odb::query<cc::model::CppMemberType>::kind == model::CppMemberType::Kind::Field))
+      {
+        typesFound.insert(member.memberTypeHash);
+      }
```

Comment:
My current solution is not complete. This coupling is not only defined by member fields but also by function parameters, inheritance relations etc. Please complement this calculation accordingly.